### ISSUE TITLE
Getter returning `this` type guard does not work

### ISSUE THREAD

**Issue by masaeedu** [Microsoft/TypeScript#6994]
It would be useful for getter properties to be able to use `this` type guards:

```typescript
interface A { a: string; }
class B {
    get isA(): this is A { ... }
}

var b = new B();

if (b.isA) {
    let x = b.a; // Compile error
}
```

**Comment by mhegazy** [comment]
getters are just like properties from the typesystem view. at the end of the day, `this is A` is just a boolean. it is not clear what it means when you have these predicates on types and you are comparing them.

**Comment by mhegazy** [comment]
closing for now, we can revsit this in the future.

**Comment by masaeedu** [comment]
For when you revisit this, the use case is flag properties like `_isScalar` in the [RxJS codebase](https://github.com/ReactiveX/RxJS/blob/14afeb6571288251174bffa95ff48b765c8f1ac9/src/Observable.ts#L66). If `_isScalar` is true, the `Observable<T>` is actually a `ScalarObservable<T>`. It would be useful to avoid casts to `<ScalarObservable<T>>foo` in contexts where `foo._isScalar` is known to be true.
